composer/src/lib.rs

```diff
@@ -118,6 +118,7 @@ pub struct ComposerBasePackage {
     pub repositories: Option<Vec<ComposerRepository>>,
     pub require: Option<HashMap<String, String>>,
     pub require_dev: Option<HashMap<String, String>>,
+    #[serde_as(as = "Option<HashMap<_, OneOrMany<_, PreferOne>>>")]
```

Comment:
Had to go searching for [the docs](https://docs.rs/serde_with/3.7.0/serde_with/struct.OneOrMany.html) to understand this annotation. It seems like this would cover your requirement. Is there an existing test that verifies this or do you plan on adding one @dzuelke?

Reply:
I added a little something, @colincasey, but the `composer` sub-crate will eventually get a whole battery of tests (gotta generate all the fixtures first, it's... a lot!)